components/datetime/src/any/datetime.rs

```diff
@@ -166,7 +171,11 @@ impl DateTimeFormatter {
             calendar::load_lengths_for_any_calendar_kind(&crate::provider::Baked, locale, kind)?,
             locale,
             &options,
-        )?;
+        )
+        .map_err(|e| match e {
+            PatternForLengthError::Data(e) => DateTimeError::Data(e),
+            PatternForLengthError::Pattern(e) => DateTimeError::Pattern(e),
+        })?;
 
```

Comment:
Suggestion: I don't think there is much benefit from having `GetSymbolForEraError`, `UnsupportedOptionsOrDataOrPatternError`, etc. These are all internal types. The user-facing one, which comes via TryWriteable, should be `DateTimeWriteError`, and all the internal helper functions can just return variants of that error type rather than each having their own error.

Reply:
I think there's a big value for maintenance. When refactoring code, you don't have to dive into each leaf call to figure out which error variants are actually possible.

Reply:
ok I'm convinced